web/src/js/app.tsx

```diff
@@ -5,19 +5,12 @@ import { Provider } from "react-redux";
 import ProxyApp from "./components/ProxyApp";
 import { add as addLog } from "./ducks/eventLog";
 import useUrlState from "./urlState";
-import WebSocketBackend from "./backends/websocket";
-import StaticBackend from "./backends/static";
 import { store } from "./ducks";
+import BackendSingleton from "./backends/BackendSingleton";
 
 useUrlState(store);
-// @ts-expect-error custom property on window
-if (window.MITMWEB_STATIC) {
-    // @ts-expect-error new property on window for debugging
-    window.backend = new StaticBackend(store);
-} else {
-    // @ts-expect-error new property on window for debugging
-    window.backend = new WebSocketBackend(store);
-}
```

Comment:
This already was a global static instance / singleton, no? I don't quite understand what problem we are solving here.

We can turn this into `export const backend = ...` followed by `window.backend = backen`. Then you can just import `backend`. No need for Java-isms. :)